packages/components/src/Typography/Typography.tsx

```diff
@@ -51,18 +53,26 @@ export function Typography({
   textCase,
   textColor,
   emphasisType,
+  numberOfLines,
 }: TypographyProps) {
+  const shouldTruncateText = numberOfLines && numberOfLines > 0;
   const className = classnames(
     styles.base,
     fontWeights[fontWeight],
     size && fontSizes[size],
     textCase && textCases[textCase],
     textColor && textColors[textColor],
     emphasisType && emphasis[emphasisType],
+    shouldTruncateText && truncate.textTruncate,
   );
 
+  const truncateLines: CSSProperties = {};
+  if (shouldTruncateText) {
+    truncateLines.WebkitLineClamp = numberOfLines;
+  }
+
   return (
-    <Tag id={id} className={className}>
+    <Tag id={id} className={className} style={truncateLines}>
```

Comment:
Sorry, I am not sure how to conditionally add a prop. It is because of this that lot of snapshot tests are updated.